File: Programmers/Level_1/둘만의 암호.js
function solution(s, skip, index) {
    var answer = "";

    var alphabet = [..."abcdefghijklmnopqrstuvwxyz"];
    var skip_index = [];

    for (var i = 0; i < skip.length; i++) {
        skip_index.push(alphabet.indexOf(skip[i]));
    }

    for (var i = 0; i < s.length; i++) {
        var s_index = alphabet.indexOf(s[i]);

        var count = 0;

        while (count !== index) {
            if (s_index + 1 < alphabet.length) {
                s_index++;
            } else {
                s_index = 0;
            }

            if (skip_index.indexOf(s_index) != -1) {
                count--;
            }

            count++;
        }

        answer += alphabet[s_index];
    }

    return answer;
}

File: Programmers/Level_1/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string s, string skip, int index)
{
    string answer = "";

    vector<char> alphabet;

    for (int i = 97; i <= 122; i++)
    {
        alphabet.push_back(i);
    }

    vector<int> skip_index;

    for (int i = 0; i < skip.size(); i++)
    {
        skip_index.push_back(find(alphabet.begin(), alphabet.end(), skip[i]) - alphabet.begin());
    }

    for (int i = 0; i < s.size(); i++)
    {
        int s_index = find(alphabet.begin(), alphabet.end(), s[i]) - alphabet.begin();
        int count = 0;

        while (count != index)
        {

            if (s_index + 1 < alphabet.size())
            {
                s_index++;
            }
            else
            {
                s_index = 0;
            }

            if (find(skip_index.begin(), skip_index.end(), s_index) != skip_index.end())
            {
                count--;
            }

            count++;
        }

        answer += (alphabet[s_index]);
    }

    return answer;
}